[Feature:RainbowGrades] Make local_pull, local_push command

ziesski opened the pull request:
<img width="861" alt="스크린샷 2024-06-14 오후 3 54 49" src="https://github.com/Submitty/RainbowGrades/assets/123261952/efc2a4a1-2fda-45a1-aca5-b4006b88ead8">


Currently make pull & push faces some rsync error as shown on screenshot above. 
For reference, it used to run with root, we just had to set password for submitty_daemon. 


From dev env, it is basically copying the directories, so new commands cp_pull & cp_push are introduced.

bmcutler commented:
Looks good.  Just do a bit of stress testing and make sure the copy works repeatedly (that it doesn't make nested copies of the files).  Also, consider switching to rsync (will be a little faster if the files already exist or only need minor updates).

ziesski commented:
> Looks good. Just do a bit of stress testing and make sure the copy works repeatedly (that it doesn't make nested copies of the files). Also, consider switching to rsync (will be a little faster if the files already exist or only need minor updates).

Renamed the cp_push and cp_pull to local_push & local_pull as we are no longer using cp command. Updated the title accordingly for future reference purposes. 

switched to rsync, and used -azPq flag, azP is same as the original make pull & push, but added q flag is for suppressing the output messages. Otherwise each json and html files names will be printed which is a lot. But @bmcutler if you disagree and think we should output all information, feel free to edit my commit (just remove q flag) and merge. 
I am adding screenshot of both with and without the q flag. 

<img width="742" alt="스크린샷 2024-06-14 오후 11 53 47" src="https://github.com/Submitty/RainbowGrades/assets/123261952/d3cfa006-2890-4479-9c02-8a5eef83acf5">

<img width="1028" alt="스크린샷 2024-06-14 오후 11 53 28" src="https://github.com/Submitty/RainbowGrades/assets/123261952/117d3458-176d-4067-b032-1981ed342df3">
